### ISSUE TITLE
No rule for sequential `write` calls (`for-loop-writes (FURB122)]` matches loops only)

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#24259]
### Summary

[for-loop-writes (FURB122)](https://docs.astral.sh/ruff/rules/for-loop-writes/#for-loop-writes-furb122) says 
> When writing a batch of elements, it's more idiomatic to use a single method call to IOBase.writelines, rather than write elements one by one.

That rule checks for `write` in loop, but sequential `write` calls, which match the exact same reasoning, are not flagged.

Given the following example, should either cases be caught?

```py
import pathlib

# from
def foo(fname: str) -> None:
    with pathlib.Path(fname).open("w", encoding="utf-8") as output:
        output.write("a\n")
        output.write("b\n")
        output.write("c\n")
        for subtitle, suffix in (
            ("Overviews", "_overview"),
            ("Modules", "_modules"),
            ("Objects", "_objects"),
        ):
            output.write(f"{suffix} {subtitle}\n")
        output.write("d\n")

# to
def foo(fname: str) -> None:
    with pathlib.Path(fname).open("w", encoding="utf-8") as output:
        output.writelines(("a\n", "b\n", "c\n"))
        output.writelines(
            f"{suffix} {subtitle}\n"
            for subtitle, suffix in (
                ("Overviews", "_overview"),
                ("Modules", "_modules"),
                ("Objects", "_objects"),
            )
        )
        output.write("d\n")

# or even, if we take collection-literal-concatenation (RUF005) into account
def foo(fname: str) -> None:
    with pathlib.Path(fname).open("w", encoding="utf-8") as output:
        output.writelines(
            [
                "a\n",
                "b\n",
                "c\n",
                *[
                    f"{suffix} {subtitle}\n"
                    for subtitle, suffix in (
                        ("Overviews", "_overview"),
                        ("Modules", "_modules"),
                        ("Objects", "_objects"),
                    )
                ],
                "d\n",
            ]
        )
```

Playground: https://play.ruff.rs/18194c34-af39-462b-9633-0c0af9281da8
